src/MgvOfferTaking.sol

```diff
@@ -306,146 +306,97 @@ abstract contract MgvOfferTaking is MgvHasOffers {
-          continue;
-        } else {
-          {
-            Tick tick = Tick.wrap(int(targets[i][1]));
-            require(TickLib.inRange(tick), "mgv/snipes/tick/outOfRange");
-            mor.maxTick = tick;
-          }
-          {
-            uint fillVolume = targets[i][2];
-            require(uint96(fillVolume) == fillVolume, "mgv/snipes/volume/96bits");
-            mor.fillVolume = fillVolume;
-          }
+      emit OrderStart();
 
-          /* We start be enabling the reentrancy lock for this (`outbound_tkn`,`inbound_tkn`) pair. */
-          sor.local = sor.local.lock(true);
-          pair.local = sor.local;
+      /* We start be enabling the reentrancy lock for this (`outbound_tkn`,`inbound_tkn`) pair. */
+      sor.local = sor.local.lock(true);
+      pair.local = sor.local;
 
-          /* `execute` will adjust `sor.wants`,`sor.gives`, and may attempt to execute the offer if its price is low enough. It is crucial that an error due to `taker` triggers a revert. That way [`mgvData`](#MgvOfferTaking/statusCodes) not in `["mgv/tradeSuccess","mgv/notExecuted"]` means the failure is the maker's fault. */
-          /* Post-execution, `sor.wants`/`sor.gives` reflect how much was sent/taken by the offer. */
-          (uint gasused, bytes32 makerData, bytes32 mgvData) = execute(pair, mor, sor);
+      {
+        /* `execute` will adjust `sor.wants`,`sor.gives`, and may attempt to execute the offer if its price is low enough. It is crucial that an error due to `taker` triggers a revert. That way [`mgvData`](#MgvOfferTaking/statusCodes) not in `["mgv/tradeSuccess","mgv/notExecuted"]` means the failure is the maker's fault. */
+        /* Post-execution, `sor.wants`/`sor.gives` reflect how much was sent/taken by the offer. */
+        (uint gasused, bytes32 makerData, bytes32 mgvData) = execute(pair, mor, sor);
 
-          if (mgvData == "mgv/tradeSuccess") {
-            successCount += 1;
-          }
+        require(mgvData != "mgv/tradeSuccess", "mgv/clean/offerDidNotFail");
+        require(mgvData != "mgv/notExecuted", "mgv/clean/tickTooLow");
 
-          /* In the market order, we were able to avoid stitching back offers after every `execute` since we knew a continuous segment starting at best would be consumed. Here, we cannot do this optimisation since offers in the `targets` array may be anywhere in the book. So we stitch together offers immediately after each `execute`. */
-          if (mgvData != "mgv/notExecuted") {
-            // updates best&tick
-            sor.local = dislodgeOffer(pair, sor.offer, sor.local, true);
-          }
+        /* In the market order, we were able to avoid stitching back offers after every `execute` since we knew a continuous segment starting at best would be consumed. Here, we cannot do this optimisation since offers in the `targets` array may be anywhere in the book. So we stitch together offers immediately after each `execute`. */
```

Comment:
also a comment that needs to be updated. You added a FIXME further up, maybe also here? Though I suspect the code has quite a bit of outdated comments now.

Reply:
All of the comments in that file should be updated :-) I made a general comment on this (and @adhusson added a specific one) and I think it'll be easiest to do it in one go once the `feat/ticks` has stabilized. But I can be persuaded otherwise :-)